src/sankey_diagram_plotly.py

```diff
@@ -401,63 +312,28 @@ def add_node(node_name):
-
-logger.info(f"Repositories Clone Directory : {working_directory / clone_directory}")
+input_file_path = working_directory / input_file
+logger.info(f"Input file path: {input_file_path}")
+try:
+    df = load_data(working_directory / input_file)
+except FileNotFoundError:
+    logger.error("Input file not found at: " + str(input_file_path))
+    sys.exit(1)
+except pd.errors.EmptyDataError:
+    logger.error("Input file is empty at: " + str(input_file_path))
+    sys.exit(1)
+except Exception as e:
+    logger.error(f"An error occurred while loading the data: {str(e)}")
+    sys.exit(1)
+
+
+logger.info(f"Repositories Clone Directory: {working_directory / clone_directory}")
 beginning_clone_dir = str(working_directory / clone_directory) + "/"
-logger.info(f"Beginning_clone_dir : {beginning_clone_dir}")
-# Initialise columns
-logger.info(f"Creating and initialising columns : {test_runners.keys()}")
-for runner in test_runners.keys():
-    df[f"{runner}_dependency_patterns"] = 0
-    df[f"{runner}_config_files"] = 0
-    df[f"{runner}_file_patterns"] = 0
-
-logger.info(
-    'Counting test runner files matching "dependency_patterns", '
-    '"config_files", "file_patterns" and saving the values in the '
-    "respective columns"
-)
-
-for index, row in df.iterrows():
-    if row["clone_status"] == "successful":
-        parts = row["repourl"].split("/")
-        clone_dir = (
-            beginning_clone_dir
-            + sanitise_directory_name(row["repodomain"])
-            + "/"
-            + parts[-1]
-        )
-
-        # runner_presence = measure_performance(detect_test_runners2, clone_dir)
-        # Measure the performance of the detect_test_runners function
-        # runner_presence = measure_performance(detect_test_runners, clone_dir)
-        runner_presence = detect_test_runners2(clone_dir)
-
-        for runner, details in runner_presence.items():
-            df.at[index, f"{runner}_dependency_patterns"] = details[
-                "dependency_patterns"
-            ]
-            df.at[index, f"{runner}_config_files"] = details["config_files"]
-            df.at[index, f"{runner}_file_patterns"] = details["file_patterns"]
-
-
-# Convert performance records to a DataFrame for easy manipulation and analysis
-# performance_df = pd.DataFrame(performance_records)
-# performance_df.to_csv(working_directory / "data" /
-#                       "performance_df_file_opening_first.csv",
-#                       index=False)
-# Calculate summary statistics
-# summary_stats = performance_df.describe()
-# print(summary_stats)
-
-logger.info(f"Sankey input dataframe saved in: " f"{working_directory/output_file}")
-df.to_csv(working_directory / output_file, index=False)
+logger.info(f"Beginning_clone_dir: {beginning_clone_dir}")
 
 node_dict, sources, targets, values = prepare_sankey_data(df)
 
-# Visualise the Sankey Diagram including the Duplicates layer
+# Visualize the Sankey Diagram including the Duplicates layer
```

Comment:
I prefer the British spelling i.e. Visualise :)

Reply:
Sure. This is done.